NullReferenceException in CodeGeneration

```
1>------ Build started: Project: UBS, Configuration: Debug Any CPU ------
1>Orleans.CodeGenerator - command-line = SourceToSource C:\projects\UBS\UBS\obj\Debug\net6.0\win-x64\UBS.orleans.g.args.txt
1>-- Code Generation FAILED --
1>
1>Exc level 0: System.NullReferenceException: Object reference not set to an instance of an object.
1>   at Orleans.CodeGenerator.Utilities.SymbolExtensions.GetAttributes(ISymbol symbol, INamedTypeSymbol attributeType, AttributeData[]& attributes) in /_/src/Orleans.CodeGenerator/Utilities/SymbolExtensions.cs:line 176
1>   at Orleans.CodeGenerator.Analysis.CompilationAnalyzer.<Analyze>g__ExpandKnownAssemblies|34_0(IAssemblySymbol asm) in /_/src/Orleans.CodeGenerator/Analysis/CompilationAnalyzer.cs:line 321
1>   at Orleans.CodeGenerator.Analysis.CompilationAnalyzer.<Analyze>g__ExpandKnownAssemblies|34_0(IAssemblySymbol asm) in /_/src/Orleans.CodeGenerator/Analysis/CompilationAnalyzer.cs:line 319
1>   at Orleans.CodeGenerator.Analysis.CompilationAnalyzer.Analyze() in /_/src/Orleans.CodeGenerator/Analysis/CompilationAnalyzer.cs:line 279
1>   at Orleans.CodeGenerator.CodeGenerator.AnalyzeCompilation() in /_/src/Orleans.CodeGenerator/CodeGenerator.cs:line 66
1>   at Orleans.CodeGenerator.CodeGenerator.GenerateCode(CancellationToken cancellationToken) in /_/src/Orleans.CodeGenerator/CodeGenerator.cs:line 53
1>   at Orleans.CodeGenerator.MSBuild.CodeGeneratorCommand.Execute(CancellationToken cancellationToken) in D:\build\agent\_work\47\s\src\Orleans.CodeGenerator.MSBuild\CodeGeneratorCommand.cs:line 143
1>   at Microsoft.Orleans.CodeGenerator.MSBuild.Program.SourceToSource(String[] args) in D:\build\agent\_work\47\s\src\Orleans.CodeGenerator.MSBuild\Program.cs:line 139
1>   at Microsoft.Orleans.CodeGenerator.MSBuild.Program.Main(String[] args) in D:\build\agent\_work\47\s\src\Orleans.CodeGenerator.MSBuild\Program.cs:line 26
1>C:\Users\Jerry.Wang\.nuget\packages\microsoft.orleans.codegenerator.msbuild\3.6.0\build\Microsoft.Orleans.CodeGenerator.MSBuild.targets(123,5): error MSB3073: The command ""dotnet" "C:\Users\Jerry.Wang\.nuget\packages\microsoft.orleans.codegenerator.msbuild\3.6.0\build\..\tasks\Orleans.CodeGenerator.MSBuild.dll" SourceToSource "C:\projects\UBS\UBS\obj\Debug\net6.0\win-x64\UBS.orleans.g.args.txt"" exited with code 3.
1>Done building project "UBS.csproj" -- FAILED.
```

--- a/src/Orleans.CodeGenerator/Analysis/CompilationAnalyzer.cs
+++ b/src/Orleans.CodeGenerator/Analysis/CompilationAnalyzer.cs
@@ -285,6 +285,13 @@ namespace Orleans.CodeGenerator.Analysis
 
             void ExpandKnownAssemblies(IAssemblySymbol asm)
             {
+                // This can occur if a developer adds [assembly: KnownAssembly(type(x))] where x is the main assembly being inspected
+                // by the code generator. i.e., the attribute is self-referential.
+                if (asm is null)
+                {
+                    return;
+                }
+
                 if (!this.KnownAssemblies.Add(asm))
                 {
                     return;

